### PR TITLE
do_print_ex(): Avoid possible integer overflow

### PR DESCRIPTION
**Pull request by t8m**
Fixes Coverity 1604657
Fixes openssl/project#780

### CODE REVIEW

**Review by nhorman — APPROVED** [review]

File: crypto/asn1/a_strex.c [hunk 1]
```diff
@@ -343,8 +344,13 @@ static int do_print_ex(char_io *io_ch, void *arg, unsigned long lflags,
...
     if (lflags & ASN1_STRFLGS_SHOW_TYPE) {
         const char *tagname;
+        size_t tagname_len;
+
         tagname = ASN1_tag2str(type);
-        outlen += strlen(tagname);
+        tagname_len = strlen(tagname);
+        if (tagname_len > INT_MAX - 1)
+            return -1;
+        outlen += (int)tagname_len;
```
Comment (tom-cosgrove-arm) [inline crypto/asn1/a_strex.c:353]: Damn, `outlen` and the return values from these functions should really be `size_t` :(

**Review by tom-cosgrove-arm — APPROVED** [review]

File: crypto/asn1/a_strex.c [hunk 2]
```diff
@@ -343,8 +344,13 @@ static int do_print_ex(char_io *io_ch, void *arg, unsigned long lflags,
 
     if (lflags & ASN1_STRFLGS_SHOW_TYPE) {
         const char *tagname;
+        size_t tagname_len;
+
         tagname = ASN1_tag2str(type);
-        outlen += strlen(tagname);
+        tagname_len = strlen(tagname);
+        if (tagname_len > INT_MAX - 1)
```
Comment (mattcaswell) [inline crypto/asn1/a_strex.c:351]: This is a bit silly really. `tagname_len` will never be greater than 17 as far as I can tell. Can't we just do the cast to `int` unconditionally?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: crypto/asn1/a_strex.c (hunk 2 above)
Comment (t8m) [reply crypto/asn1/a_strex.c]: fixup pushed

**Comment by t8m** [comment]
@nhorman @tom-cosgrove-arm please reconfirm

**Comment by nhorman** [comment]
ack, approval holds

**Review by tom-cosgrove-arm — APPROVED** [review]
reconfirming

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to all the active branches. Thank you for the reviews.

**Pull request closed without merging**